src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateResolver.cs

```diff
@@ -269,30 +277,23 @@ public static IReadOnlyCollection<ITemplateMatchInfo> PerformCoreTemplateQuery(I
             )
             .Where(x => !IsTemplateHiddenByHostFile(x.Info, hostDataLoader)).ToList();
 
+            //select only the templates which do not have mismatches
             IReadOnlyList<ITemplateMatchInfo> coreMatchedTemplates = templates.Where(x => x.IsMatch).ToList();
-
-            if (coreMatchedTemplates.Count == 0)
-            {
-                // No exact matches, take the partial matches and be done.
-                coreMatchedTemplates = templates.Where(x => x.IsPartialMatch).ToList();
```

Comment:
Before we had logic that fall back to partial matches if there was no exact matches, is that logic now gone intentionally?

Reply:
Add unit test when typing `dotnet new Application`, it should show this output(preferable without --language comment since it wouldn't solve this problem):
![image](https://user-images.githubusercontent.com/774791/102450807-8db8a780-4037-11eb-85f1-ee9fac1ad1ee.png)

Reply:
> Before we had logic that fall back to partial matches if there was no exact matches, is that logic now gone intentionally?

That is intentional. Probably we can check it with Kathleen once again. Previously we removed such logic for list and help. 
This logic allows for example the following: given you have 2 templates installed foo F# and bar C# and `dotnet new blabla --language C#`, bar C# will be instantiated. 
We limit the list and help to show the templates only if there is at least partial match for name.

Reply:
> Add unit test when typing `dotnet new Application`, it should show this output(preferable without --language comment since it wouldn't solve this problem):

I think that's correct output, using exact short name will solve the issue. However since I've added separate status for language ambiguity we can split error message and show 
- `use exact shore name` in this case
- `select the language to use with --language option` for language ambiguity case

Reply:
Oh, I gave wrong sample, I meant "dotnet new common"